test/test_endpoints.py

```diff
@@ -988,7 +1016,7 @@ def test_illegal_arg_ephemeral_auth(self, ftb):
         with self.assertRaises(ValueError) as ctx:
             TCPHiddenServiceEndpoint(
                 reactor, config, 80,
-                stealth_auth=['alice', 'bob'],
+                auth=AuthStealth(['alice', 'bob']),
```

Comment:
This turned out not to be necessary.  I thought this test was failing at one point but it turned out that a Deferred with an unhandled failure had leaked from a previous test and only happened to be gc'd as this one was running.  Before I noticed that I updated the test to use the non-deprecated codepath to make it easier to follow the logic in `TCPHiddenServiceEndpoint.__init__`.